File: spring-boot-backend/src/main/java/com/eafc/springbootbackend/entities/shopping/OrderItem.java
package com.eafc.springbootbackend.entities.shopping;

import com.eafc.springbootbackend.entities.product.ProductInfo;
import com.eafc.springbootbackend.entities.product.Stock;
import lombok.Data;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Data
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long orderItemId;

    @ManyToOne
    @JoinColumn(name = "product", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private ProductInfo product;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "orderInfo")
    private OrderInfo order;

    @NotNull
    private double purchasedPrice;

    @NotNull
    private double totalPrice;

    @OneToOne
    private Stock relatedStockAndSize;

    @NotNull
    private int quantity;
}

File: angular-frontend/src/app/common/users/order-item.ts
import {ProductInfo} from "../shopping/product-info";
import {OrderInfo} from "./orderInfo";
import {Stock} from "../prod-details/stock";

export class OrderItem {

  orderItemId!: number;
  product!: ProductInfo;
  order!: OrderInfo;
  purchasedPrice!: number;
  size!: string;
  quantity!: number;
  relatedStockAndSize!: Stock;
}
